packages/lib-user/src/components/UserHome/components/Dashboard/DashboardContainer.js

```diff
@@ -0,0 +1,40 @@
+import { panoptes } from '@zooniverse/panoptes-js'
+import auth from 'panoptes-client/lib/auth'
+import useSWR from 'swr'
+
+import Dashboard from './Dashboard'
+
+const SWROptions = {
+  revalidateIfStale: true,
+  revalidateOnMount: true,
+  revalidateOnFocus: true,
+  revalidateOnReconnect: true,
+  refreshInterval: 0
+}
+
+/* This is a similar pattern to usePanoptesUser hook, but includes the profile_header */
+const fetchProfileBanner = async ({ authUser }) => {
+  const token = await auth.checkBearerToken()
+  const authorization = `Bearer ${token}`
+  const query = { include: 'profile_header', user_id: authUser.id }
+
+  try {
+    const { body } = await panoptes.get('/users', query, { authorization })
+    const user = body.users?.[0]
```

Comment:
This API request runs a search across the Users table, returning all users that match `query`. If the user ID is undefined, it will return the first 20 users from the Users table. 

When you want a single, specific resource by ID, the form of the URL to use is `GET /api/users/1234`. 

You don't need an Authorization header here either, since users are public resources. 

When you add Authorization headers to a request, the browser generally won't cache the response, so you might end up generating unnecessary network traffic. That can become an important consideration when you're building a busy page with a lot of users, like the Zooniverse home page.

Reply:
Good to know! Thanks, I'll refactor this fetch.

Reply:
Yes. A unique cache key for this URL is:

```js
const key = {
  endpoint: `/users/{authUser.id}`,
  query: { includes: ‘profile_header’ }
}
```

With that key, the client request becomes:

```js
panoptes.get(key.endpoint, key.query)
```

Alternatively, you could set the key as the full URL:

```js
const key = `/users/{authUser.id}?include=profile_header`
…
panoptes.get(key)
```

Reply:
Sorry, one other thing. `panoptes.get` uses an old version of SuperAgent (I think it’s still on v8?), which means that these requests aren’t using `fetch`. 

The app router in Next.js is built on top of `fetch` so code like this won’t be able to take advantage of Next’s built-in data-caching. Not a big deal right now, when you’re testing with just a single visitor, but something to have in mind when you roll out and need to run the home page under load.

Reply:
How does Next.js's built-in data-caching play with SWR? This request is using SWR in a client component, which is in contrast to using `fetch` directly in a page.js in app-root:
https://github.com/zooniverse/front-end-monorepo/blob/9a7a630f69c338ee941cfeb4c79550eb9099251a/packages/app-root/src/app/page.js#L22-L24

(Linking the existing superagent Issue to this comment: https://github.com/zooniverse/front-end-monorepo/issues/317)

Reply:
In the app router, with Next.js 13.5 and higher, Vercel replace native  `fetch` with their own global `fetch` that caches the return value.

Reply:
SWR can be used with `fetch`, like `react-query`, but will work with any function that returns a Promise-like object.

Reply:
Here’s some documentation on data caching with libraries that don’t use `fetch`, such as `panoptes-js`.
https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-third-party-libraries

Here’s documentation for the app router data cache, which you’ll need to manage across app instances and deployments, since Zooniverse Next.js is self-hosted on Node. 
https://nextjs.org/docs/app/building-your-application/caching#data-cache

Reply:
Here’s a couple of old issues about slow page build times, for context. 

- #2741. 
- #3341.